src/Workspaces/Core/Portable/Shared/Collections/IntervalTree`2.cs

```diff
@@ -17,13 +17,14 @@ namespace Microsoft.CodeAnalysis.Shared.Collections
     /// [start, end).  It allows you to efficiently find all intervals that intersect or overlap 
     /// a provided interval.
     /// </summary>
-    internal partial class IntervalTree<T> : IEnumerable<T>
+    internal partial class IntervalTree<T, TIntrospector> : IEnumerable<T>
+        where TIntrospector : struct, IIntervalIntrospector<T>
```

Comment:
instead of the instrospector being defined here as a type param.  woudln't it be cleaner to be a generic type param to methods like IntersectsWith?

Reply:
another way to put ths.  the introspector i never state of the interval tree. so it doesn't really need to be part of its declaration type.

Reply:
Making it a generic type means the JIT can inline its implementation.

Reply:
i'll whip up a PR to show you what i mean.  it would still be generic and inlinable.  just at the method level (like shapes) not at teh type-decl level.

Reply:
If this was a public API, I'd be learning more towards that approach. For an internal API, I'm not sure it makes a notable difference. The sample change adds more code than it removes.

Reply:
I restored `IntervalTree<T>` to only needing one generic type argument.

Reply:
great, thanks :)